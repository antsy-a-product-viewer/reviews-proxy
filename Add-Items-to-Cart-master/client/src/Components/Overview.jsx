import React from 'react';
import OverviewList from './OverviewList.jsx';
import style from './style.css.js'

class Overview extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      overview: [],
      hasOverview: false,
    };
    this.getOverview = this.getOverview.bind(this);
  }

  componentDidMount() {
    this.getOverview();
  }

  getOverview() {
    let overviewBulletPoints = [];
    let overviewLength = false;
    const overview = this.props.data[0].overview;
    if (overview.length > 0) {
      overviewLength = true;
      overviewBulletPoints = overview;
    }
    this.setState({
      overview: overviewBulletPoints,
      hasOverview: overviewLength,
    });
  }

  render() {
    return (
      <div>

        {this.state.hasOverview
          ? (
            <div className="OverallOverview">
              <div className="Overview" style={style.bold}>
            Overview
              </div>
              <OverviewList overviews={this.state.overview} />
              <br />
              <hr style={style.hr}/>
              {' '}
            </div>
          )
          : null }

      </div>
    );
  }
}


export default Overview;
