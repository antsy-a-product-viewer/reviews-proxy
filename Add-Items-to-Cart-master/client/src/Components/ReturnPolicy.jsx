import React from 'react';
import Modal from 'react-modal';
import style from './style.css.js'

const customStyles = {
  content: {
    top: '50%',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)',
  },
};

class ReturnPolicy extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      exclusions: [],
      returnPolicy: [],
      returnOverview: [],
      giftWrapping: [],
      returnPolicyShown: false,
    };
    this.getReturnPolicy = this.getReturnPolicy.bind(this);
    this.showReturnPolicy = this.showReturnPolicy.bind(this);
    this.closeReturnPolicy = this.closeReturnPolicy.bind(this);
  }

  componentDidMount() {
    this.getReturnPolicy();
  }

  getReturnPolicy() {
    let returnable;
    let returnPolicy = '';
    let exclusion;
    let wrapping = '';
    console.log(this.props.data[0])
    if (this.props.data[0].availableToReturn) {
      returnable = 'Returns and exchanges accepted';
      exclusion = 'Exceptions may apply. ';
      returnPolicy = 'See return policy';
      wrapping = 'Gift wrapping available';
    } else {
      returnable = 'No returns or exchanges';
      exclusion = 'But please contact me if you have any problems with your order.';
    }
    this.setState({
      exclusions: exclusion,
      returnPolicy: returnable,
      returnOverview: returnPolicy,
      giftWrapping: wrapping,
    });
  }

  showReturnPolicy() {
    this.setState({
      returnPolicyShown: true,
    });
  }

  closeReturnPolicy() {
    this.setState({
      returnPolicyShown: false,
    });
  }


  render() {
    return (
      <div>
        <div className="returnOverview">
          <div className="returns" style={style.bold}>
            {this.state.returnPolicy}
          </div>
          <div className="returnPolicy">
            <span>{this.state.exclusions}</span>
            <span
              className="Return"
              style={style.cursor}
              onClick={this.showReturnPolicy}
            >
              {this.state.returnOverview}
            </span>
            <Modal
              isOpen={this.state.returnPolicyShown}
              onRequestClose={this.closeReturnPolicy}
              style={customStyles}
            >
              <div className="modalContainer" style={style.returnModal}>
                <h1>Returns & Exchanges</h1>
                <p> I gladly accept returns, exchanges, and cancellations</p>
                <p>Contact me within: 14 days of delivery </p>
                <p>Ship items back within: 30 days of delivery</p>
                <p>Request a cancellation within: 2 days of purchase</p>
                <h2>The following items can't be returned or exchanged</h2>
                <p>Because of the nature of these items, unless they arrive damaged or defective, I can't accept returns for: </p>
                <li style={style.bulletPoint}>Custom or personalized orders</li>
                <li style={style.bulletPoint}>Perishable products (like food or flowers)</li>
                <li style={style.bulletPoint}>Digital downloads</li>
                <li style={style.bulletPoint}>Intimate items (for health/hygiene reasons)</li>
              </div>
              <button className="closePolicy" onClick={this.closeReturnPolicy}> Close me </button>
            </Modal>

          </div>
          {' '}
          <br />
          <br />
          <div className="giftwrappingOptions">
            {this.state.giftWrapping}
          </div>
        </div>
        {' '}
        <hr style={style.hr}/>
      </div>
    );
  }
}

export default ReturnPolicy;
