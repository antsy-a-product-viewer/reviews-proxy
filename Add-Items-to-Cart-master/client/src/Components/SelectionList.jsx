import React from 'react';
import SelectionListPossibilities from './selectionListPossibilities.jsx';


class SelectionList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      stateChange: false,
      itemInformation: [],
    };
    this.getItemInformation = this.getItemInformation.bind(this);
  }

  componentDidMount() {
    this.getItemInformation();
  }

  getItemInformation() {
    let itemInfo = {};
    let stateChanged = false;
    if (this.props.data[0].options) {
      itemInfo = this.props.data[0].options;
      stateChanged = true;
    }
    this.setState({
      itemInformation: itemInfo,
      stateChange: stateChanged,
    });
  }

  render() {
    return (
      <div className="selectionList">
        {this.state.stateChange ? <SelectionListPossibilities selection={this.state.itemInformation} /> : null }
      </div>
    );
  }
}


export default SelectionList;
