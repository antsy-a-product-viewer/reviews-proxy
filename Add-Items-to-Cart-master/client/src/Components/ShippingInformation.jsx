import React from 'react';
import style from './style.css.js'

class ShippingInformation extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      shippingTime: [],
      getShippingCost: false,
      shippingCost: [],
      enteredShippingInfo: false,
      zipCode: [],
      shippingFrom: [],
      shippingTo: [],
      shippingToCountry: 'China',
      deliveryDate: [],
    };
    this.getShippingInformation = this.getShippingInformation.bind(this);
    this.getShippingInfo = this.getShippingInfo.bind(this);
    this.showShipping = this.showShipping.bind(this);
    this.shippingCost = this.shippingCost.bind(this);
    this.shippingToCountry = this.shippingToCountry.bind(this);
  }


  componentDidMount() {
    this.getShippingInformation();
  }

  getShippingInformation() {
    let manufacturingTime = null;
    const shippingFrom = this.props.data[0].shippingFrom;
    const shippingTo = this.props.data[0].shippingTo;
    const deliveryDate = this.props.data[0].deliveryDate;
    manufacturingTime = this.props.data[0].manufacturingTime;
    this.setState({
      shippingTime: manufacturingTime,
      shippingFrom,
      shippingTo,
      deliveryDate,
    });
  }

  getShippingInfo() {
    event.preventDefault();
    this.setState({
      getShippingCost: !this.state.getShippingCost,
    });
  }

  showShipping() {
    event.preventDefault();
  }

  shippingCost() {
    const shippingCost = this.props.data[0].shippingCosts;
    if (typeof parseInt(event.target.value) === 'number' && event.target.value.length === 5) {
      this.setState({
        shippingCost,
        enteredShippingInfo: true,
        getShippingCost: !this.state.getShippingCost,
        zipCode: event.target.value,
      });
    }
  }

  shippingToCountry() {
    this.setState({
      shippingToCountry: event.target.value,
    });
  }

  render() {
    return (
      <div>
        <div className="shippingInformation">
          <div style={style.bold}>
        Shipping & Returns
          </div>
          {!this.state.enteredShippingInfo
            ? (
              <div>
                <div className="manufacturingTime" style={style.bold}>
              Ready to ship in
                  {' '}
                  {this.state.shippingTime}
-
                  {this.state.shippingTime + 2}
                  {' '}
days
                </div>
                <div className="shippedFrom">
            From
                  {' '}
                  {this.state.shippingFrom}
                </div>
                <div className="getShippingCost" style={style.cursor} onClick={this.getShippingInfo}>
             Get Shipping Cost
                </div>
              </div>
            )

            : (
              <div>
                <div className="deliver" style={style.bold}>
            Estimated delivery:
                  {' '}
                  {this.state.deliveryDate[0]}
                  {' '}
                  {this.state.shippingTime + 2}
                  {' '}
-
                  {' '}
                  {this.state.deliveryDate[1]}
                  {' '}
                  {this.state.shippingTime + 7}
                </div>
                <div className="shippedFrom">
            From
                  {' '}
                  {this.state.shippingFrom}
                </div>
                <span className="shippingCost">
            $
                  {this.state.shippingCost}
                  {' '}
shipping to
                  {' '}
                  {' '}
                </span>
                <span
                  className="shippingAddress"
                  style={style.underline}
                  onClick={this.getShippingInfo}
                >
                  {this.state.shippingToCountry}
,
                  {' '}
                  {this.state.zipCode}
                </span>
              </div>
            )
      }

          {this.state.getShippingCost
            ? (
              <form onSubmit={this.showShipping}>
                <span style={style.leftFloat}>
              Country
                  {' '}
                  <br />
                  <select
                    className="country"
                    style={style.shippingCostInput}
                    value={this.state.value}
                    onChange={this.shippingToCountry}
                  >
                    {this.state.shippingTo.map(country => <option className="options">{country}</option>)}
                  </select>
                </span>
                <span style={style.rightFloat}>
              Zip or postal code
                  {' '}
                  <br />
                  <label>
                    <input
                      onClick={this.shippingCost}
                      id="shippingCost"
                      type="text"
                      value={this.state.value}
                    />
                  </label>
                </span>
              </form>
            ) : null}

        </div>
        {' '}
        <br />
        {' '}
        <br />
        {' '}
        <hr style={style.hr}/>
      </div>
    );
  }
}

export default ShippingInformation;
