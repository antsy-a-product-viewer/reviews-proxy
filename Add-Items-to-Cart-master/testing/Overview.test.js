import React from 'react'
import {mount, shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ItemDescription from '../client/src/Components/itemDescription.jsx';
import ShippingInformation from '../client/src/Components/ShippingInformation.jsx'
import globalFetch from 'jest-fetch-mock';
import sinon from 'sinon'
import Overview from '../client/src/Components/Overview.jsx'
import ReturnPolicy from '../client/src/Components/ReturnPolicy.jsx';
import SelectionList from '../client/src/Components/SelectionList.jsx';
import renderer from 'react-test-renderer';
configure({ adapter: new Adapter() });

const mockData = [
  {
    "overview":["short id pork picanha nostrud et deserunt mignon meatloaf est meatloaf magna "],
    "shippingTo":["China"," Wales"," Belgium"," Japan"," France"," America"," Germany"," Canada"," Spain"," Brazil"," South Africa"," Belgium"," India"," Mexico"," Argentina"," Ethiopia"," Jamaica"," Brazil"," Sweden"," Denmark"," Switzerland"," Ghana"," Bangladesh"," Switzerland"," Nigeria"," Sri Lanka"," Ireland"," Italy"," Niger"," Sudan"," Italy"," Argentina"," Pakistan"," Cuba"," North Korea"],
    "deliveryDate":["Oct","Oct"],
    "_id":"5c6511ac8f191456a859ab5f","productId":98,"price":92.65,"description":"duis filet short bacon do aliquip nulla do hock in strip nulla in andouille t-bone ribeye ","options":{"ribs":["Picanha","est","meatloaf","andouille","Andouille","steak","quis","consequat","kevin"],"pig":["boudin","Magna","beef","hock","laborum","Excepteur","mignon","id","Ham","dolore","lorem"]},
    "quantityInStock":37,
    "availableToReturn":true,
    "shippingCosts":2.99,
    "manufacturingTime":4,
    "shippingFrom":" France",
    "__v":0
  }
]


describe("Overview Component", () => {
  
  it('Should exist', () => {
    const wrapper = shallow(<ItemDescription/>);
    expect(wrapper.exists()).toBe(true)
  });
  
  it('Should render without throwing an error', () => {
    expect(shallow(<ItemDescription />).find('.description').exists()).toBe(true);
  });
  
  it('Should call componentDidMount', () => {
    const mount = jest.spyOn(ItemDescription.prototype, 'componentDidMount');
    const wrapper = shallow(<ItemDescription />);
    expect(ItemDescription.prototype.componentDidMount).toHaveBeenCalled();
    mount.mockRestore();
  });
  
  it('Should invoke functions after components mounted', () => {
    const getFindQuantity = jest.spyOn(ItemDescription.prototype, 'findQuantity');
    const getPostingDescription = jest.spyOn(ItemDescription.prototype, 'postingDescription')
    mount(<ItemDescription />);
    expect(getFindQuantity).toHaveBeenCalledTimes(1);
    expect(getPostingDescription).toHaveBeenCalledTimes(1);
  });
  
  it('Should stimulate a click event when adding to cart', () => {
    const onButtonClick = sinon.spy(ItemDescription.prototype, 'addedToCart');
    const wrapper = shallow(<ItemDescription addedToCart={onButtonClick}/>);
    wrapper.find('.addToCart').simulate('click', { preventDefault() {} });
    expect(onButtonClick.calledOnce).toBe(true)
  });

  it('Should open the modal when its state is opened', () => {
    const props = {askedQuestion: true};
    const wrapper = shallow(<ItemDescription {...props}/>);
    wrapper.update();
    expect(wrapper.find('.questionsModal').exists()).toEqual(true)
  })
})

// describe('AddToCart', () => {
//   const cart = renderer
//     .create(<Cart />)
//     .toJSON();
//   expect(cart).toMatchSnapshot(); 
// })

describe('Return Policy', () => {
  it('Should exist', () => {
    // const mount = jest.spyOn(ReturnPolicy.prototype, 'componentDidMount');
    const wrapper = shallow(<ReturnPolicy data={mockData}/>);
    expect(wrapper.exists()).toBe(true)
    // mount.mockRestore();    
  })
  it('Should open the Return Policy on click', () => {
    jest.spyOn(ReturnPolicy.prototype, 'showReturnPolicy');
    const wrapper = shallow(<ReturnPolicy data={mockData}/>);
    wrapper.find('.Return').simulate('click');
    expect(ReturnPolicy.prototype.showReturnPolicy).toHaveBeenCalled();
  })
  it('Should close the return policy on click', () => {
    jest.spyOn(ReturnPolicy.prototype, 'closeReturnPolicy');
    const wrapper = shallow(<ReturnPolicy data={mockData} />);
    wrapper.find('.Return').simulate('click');
    wrapper.find('.closePolicy').simulate('click');
    expect(ReturnPolicy.prototype.closeReturnPolicy).toHaveBeenCalled();
  })
})

describe('Selection List', () => {
  it('Should run component did mount', () => {
    const mounted = jest.spyOn(SelectionList.prototype, 'componentDidMount')
    const wrapper = mount(<SelectionList data={mockData}/>);
    expect(SelectionList.prototype.componentDidMount).toHaveBeenCalled();
    mounted.mockRestore();
  });
  it('Should return an empty div if there are no state changes', () => {
    const wrapper = shallow(<SelectionList data={mockData}/>);
    expect(wrapper.find('.selectionList').length).toEqual(1);
  });

describe('Overview', () => {
  it ('should exist', () => {
    const wrapper = mount(<Overview data={mockData}/>);
    expect(wrapper.exists()).toBe(true);
  })
  it ('Should invoke componentDidMount', () => {
    const mounted = jest.spyOn(Overview.prototype, 'componentDidMount');
    const wrapper = shallow(<Overview data={mockData}/>);
    expect(Overview.prototype.componentDidMount).toHaveBeenCalled();
  })
  // it ('Should display the correct overview', () => {
  //   const wrapper = shallow(<OverviewListEntry data={mockData}/>);
  //   let overviewData = document.getElementsByClassName('overview').childNodes.item(0).nodeValue
  //   console.log(overviewData, 'data')
  //   expect(wrapper.find(overviewData)).toEqual('short id pork picanha nostrud et deserunt mignon meatloaf est meatloaf magna')
  // })
});

describe('Shipping Information', () => {
  it ('Should invoke component did mount', () => {
    const mounted = jest.spyOn(ShippingInformation.prototype, 'componentDidMount');
    const wrapper = shallow(<ShippingInformation data={mockData}/>);
    expect(ShippingInformation.prototype.componentDidMount).toHaveBeenCalled();
  });

  it ("Should show the amount of time it takes to ship if information entered", () => {
    const props = {enteredShippingInfo : true};
    const wrapper = shallow(<ShippingInformation data={mockData} {...props}/>);
    expect(wrapper.find('.manufacturingTime').length).toEqual(1);
  });

})

})
