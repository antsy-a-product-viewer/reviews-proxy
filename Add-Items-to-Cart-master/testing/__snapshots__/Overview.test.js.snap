// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  id="container"
>
  <div
    id="description"
  >
    <div>
      <div
        className="description"
        style={
          Object {
            "fontSize": "24px",
            "fontWeight": "bold",
          }
        }
      />
       
      <br />
      <div
        style={
          Object {
            "marginBottom": "18px",
          }
        }
      >
        <span
          className="price"
          style={
            Object {
              "fontSize": "24px",
              "fontWeight": "bold",
            }
          }
        >
           
          $
           
        </span>
        <span
          className="question"
        >
          <button
            className="askQuestion"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "white",
                "cursor": "pointer",
                "float": "right",
                "fontWeight": "bold",
                "height": "24px",
                "width": "120px",
              }
            }
          >
             
            Ask A Question
             
          </button>
        </span>
      </div>
      <div
        className="optionNames"
      />
       
      <div>
        <div
          id="addToCard"
        >
          <button
            className="addToCart"
            onClick={[Function]}
            style={
              Object {
                "backgroundColor": "#222222",
                "color": "white",
                "cursor": "pointer",
                "fontSize": "16px",
                "height": "35px",
                "width": "390px",
              }
            }
          >
             
            Add to Cart
             
          </button>
        </div>
        <div>
          <span
            style={
              Object {
                "float": "left",
              }
            }
          >
            <img
              src="https://d2v9y0dukr6mq2.cloudfront.net/video/thumbnail/mCz7hqj/videoblocks-shopping-trolley-cart-icon-in-and-out-animation-black-and-white_bfguek3mbz_thumbnail-full05.png"
              style={
                Object {
                  "height": "50px",
                  "width": "90px",
                }
              }
            />
          </span>
          <span
            style={
              Object {
                "float": "left",
                "fontWeight": "bold",
              }
            }
          >
            Other people want this. 
          </span>
          <span
            style={
              Object {
                "float": "left",
              }
            }
          >
             
            5
             
            people have this in their carts right now
             
          </span>
        </div>
      </div>
       
      <br />
       
      <br />
       
      <br />
      <hr />
      <div
        className="overview"
      />
      <div
        className="ShippingInfo"
      />
      <div
        className="ReturnPolicy"
      />
    </div>
  </div>
</div>
`;
